### ISSUE TITLE
Issue with codeql query execution

### ISSUE THREAD

**Issue by arunstar** [purs3lab/Argus#1]
Coming from this [github blog post](https://github.blog/2023-08-09-four-tips-to-keep-your-github-actions-workflows-secure/) This is a awesome work. 

I wanted to try this tool, but for some reason, it's not working for me

The command I'm running.  `docker-compose run argus --mode action --url https://github.com/actions-js/push.git`

Please take a look if you guys have some spare time. Thanks

Error:

```
ERROR: Classes may not extend outside their parameterised module (/root/codeql_home/codeql-repo/shared/dataflow/codeql/dataflow/internal/AccessPathSyntax.qll:186,33-57)

Traceback (most recent call last):
  File "/root/argus.py", line 84, in <module>
    main()
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/argus.py", line 78, in main
    action.run()
  File "/root/argus_components/action.py", line 59, in run
    self._run_remote_action()
  File "/root/argus_components/action.py", line 71, in _run_remote_action
    self.report = action_obj.run()
                  ^^^^^^^^^^^^^^^^
  File "/root/argus_components/plugins/ghjs_action.py", line 93, in run
    results = CodeQL.run_codeql_query(codeql_folder)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/argus_components/utils/codeql.py", line 125, in run_codeql_query
    raise Exception(f"No results folder found for {repo_path}", _extract_db_error(repo_path, "queries"))
Exception: ('No results folder found for /tmp/actions-js#push.git_codeql', 'Check logs in log folder : /tmp/actions-js#push.git_codeql/log to parse error')
```

**Comment by jetlime** [comment]
Same issue here, the easiest would be to directly install the python package. Which is however quite annoying since the list of package requirements is missing.

**Comment by jetlime** [comment]
I have exactly the same issue as well, while using the `Argus` package with my local python distribution.
The command `python3 argus.py  --mode action --url https://github.com/actions-js/push.git` results in the error below,

```bash
2023-12-27 00:09:20,902 repo [INFO] Initialzed Action : actions-js#push.git
2023-12-27 00:09:20,902 repo [INFO] Cloning action to /tmp/actions-js#push.git
STDOUT has 
STDERR has /bin/sh: 1: /home/user/codeql_home/codeql/codeql: not found

Traceback (most recent call last):
  File "/home/user/Argus/argus.py", line 84, in <module>
    main()
  File "/usr/lib/python3/dist-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/user/Argus/argus.py", line 78, in main
    action.run()
  File "/home/user/Argus/argus_components/action.py", line 59, in run
    self._run_remote_action()
  File "/home/user/Argus/argus_components/action.py", line 71, in _run_remote_action
    self.report = action_obj.run()
  File "/home/user/Argus/argus_components/plugins/ghjs_action.py", line 84, in run
    CodeQL.compile_codeql_db(self.action_folder, codeql_folder)
  File "/home/user/Argus/argus_components/utils/codeql.py", line 89, in compile_codeql_db
    raise Exception(f"Error Creating DB : {repo_path}", _extract_db_error(repo_path))
Exception: ('Error Creating DB : /tmp/actions-js#push.git', 'No log folder found in /tmp/actions-js#push.git/log to parse error')
```

I am running the program on Ubuntu 22.04.3 LTS JAMMY, using Python 3.10.12.

**Comment by kain88-de** [comment]
I'm unable to run even the test examples. They yield the same problem. The test scripts ignore errors so this is easy to overlook in them.

**Comment by souragc** [comment]
@arunstar If you update the Dockerfile and replace line number 13 with the following lines, it will fix your issue.

```
RUN git clone https://github.com/github/codeql codeql-repo/
WORKDIR /root/codeql_home/codeql-repo
RUN git checkout 2e16b71215d8a886d8b4a41689bc58639518dcbe
WORKDIR /root/codeql_home
```

EDIT: Just saw the PR. You can also follow what is done in the PR.

**Comment by jiangnanpro** [comment]
> @arunstar If you update the Dockerfile and replace line number 13 with the following lines, it will fix your issue.
> 
> ```
> RUN git clone https://github.com/github/codeql codeql-repo/
> WORKDIR /root/codeql_home/codeql-repo
> RUN git checkout 2e16b71215d8a886d8b4a41689bc58639518dcbe
> WORKDIR /root/codeql_home
> ```
> 
> EDIT: Just saw the PR. You can also follow what is done in the PR.

Hi, I'm also trying to apply your tool using Docker, but still having similar problem even after updating the Dockerfile as you recommended. Can you guys take a look on this? Thanks!

I'm running the same command: `python3 argus.py  --mode action --url https://github.com/actions-js/push.git`

Error encountered:
```
2024-07-10 21:54:10,508 repo [ERROR] The  Command Timed out
Traceback (most recent call last):
  File "/root/argus.py", line 84, in <module>
    main()
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/argus-lKa0R1gD-py3.11/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/argus.py", line 78, in main
    action.run()
  File "/root/argus_components/action.py", line 60, in run
    self._run_remote_action()
  File "/root/argus_components/action.py", line 72, in _run_remote_action
    self.report = action_obj.run()
                  ^^^^^^^^^^^^^^^^
  File "/root/argus_components/plugins/ghjs_action.py", line 93, in run
    results = CodeQL.run_codeql_query(codeql_folder)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/argus_components/utils/codeql.py", line 125, in run_codeql_query
    raise Exception(f"No results folder found for {repo_path}", _extract_db_error(repo_path, "queries"))
Exception: ('No results folder found for /tmp/actions-js#push.git_codeql', 'Check logs in log folder : /tmp/actions-js#push.git_codeql/log to parse error')
```

**Comment by R3x** [comment]
@jiangnanpro Can you check your codeql db logs inside the docker container to see why it failed?

**Comment by jiangnanpro** [comment]
Hi, I've noticed that the problem is caused by a subprocess timeout. By manually extending the timeout limit, the QL queries can now be completed successfully. However, there's another issue that I encountered: when executing the function `save_report_to_file(self)` within `argus_components/action.py` (line 84), the program throws a KeyError, details can be found as following:
```
  File "/root/argus_components/action.py", line 84, in save_report_to_file
    file_name = RESULTS_FOLDER / f"{self.action_name}_{self.options_dict['value']}.sarif"
                                                       ~~~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'value'
```

The problem occurred when none of the optional inputs 'branch', 'commit', or 'tag' were defined while running Argus in action mode, as in the example `python3 argus.py  --mode action --url https://github.com/actions-js/push.git`. In this scenario, both `option_provided` and `option_value` will be `None`, resulting in `option_dict` being an empty dictionary, which will eventually raise the KeyError.

Maybe an if-condition should be added to the function `save_report_to_file(self)` for cases where no option is defined to prevent this issue. Or is there something that I'm missing?
